src/ucp/core/ucp_types.h

```diff
@@ -86,12 +86,12 @@ enum {
     UCP_AM_ID_ATOMIC_REQ        =  20, /* Remote memory atomic request */
     UCP_AM_ID_ATOMIC_REP        =  21, /* Remote memory atomic reply */
     UCP_AM_ID_CMPL              =  22, /* Remote memory operation completion */
-    UCP_AM_ID_SINGLE            =  23, /* For user defined Active Messages */
-    UCP_AM_ID_MULTI             =  24, /* For user defined AM if message 
-                                          does not fit in one AM */
-    UCP_AM_ID_SINGLE_REPLY      =  25, /* For user defined AM when a reply
-                                          is needed */
-    UCP_AM_ID_MULTI_REPLY       =  26,
+    UCP_AM_ID_SINGLE            =  23, /* Single fragment user defined AM */
+    UCP_AM_ID_FIRST             =  24, /* First fragment of user defined AM */
+    UCP_AM_ID_MIDDLE            =  25, /* Middle or last fragment of user
+                                          defined AM */
+    UCP_AM_ID_SINGLE_REPLY      =  26, /* Single fragment user defined AM
```

Comment:
maybe better:
```
Single user-defined AM fragment ...
```

Reply:
i'd keep it as is, because it is user defined AM, but not a user defined fragment

Reply:
I see

Reply:
can we remove UCP_AM_ID_SINGLE_REPLY and use hdr.flags instead?